lib/ts/recipe/recipeModule/utils.ts

```diff
@@ -30,7 +33,28 @@ export function normaliseRecipeModuleConfig<Action>(config: RecipeConfig<Action>
     return {
         recipeId: config.recipeId,
         appInfo: config.appInfo,
+        storageHandlerInput: config.storageHandlerInput,
         preAPIHook,
         postAPIHook,
     };
 }
+
+export function normaliseStorageHandlerInput(storageHandlerInput?: StorageHandlerInput): NormalisedStorageHandlers {
+    let localStorageFunction: StorageHandlerFunction = (original) => original;
+    let sessionStorageFunction: StorageHandlerFunction = (original) => original;
+
+    if (storageHandlerInput !== undefined) {
+        if (storageHandlerInput.localStorage !== undefined) {
+            localStorageFunction = storageHandlerInput.localStorage;
+        }
+
+        if (storageHandlerInput.sessionStorage !== undefined) {
+            sessionStorageFunction = storageHandlerInput.sessionStorage;
+        }
+    }
+
+    return {
+        localStorage: localStorageFunction(getDefaultLocalStorageHandler()),
+        sessionStorage: sessionStorageFunction(getDefaultSessionStorageHandler()),
+    };
+}
```

Comment:
Add a comment that you have intentionally not used the override builder cause you expect that the functions in the interface never call each other.